### ISSUE TITLE
Errors in relative addressing

### ISSUE THREAD

**Issue by jwusteman** [mozilla/thimble.mozilla.org#2587]
Using Chrome 62. When using Chrome dev tools in the service worker settings I disabled the "Bypass for network option" so requests always go to the service worker.
 I created a test with an index.html 2 levels deep in a sub folder, e.g. 
test/one/index.html
I added a style to the style.css file at the root:
/style.css
/* Fonts from Google Fonts - more at https://fonts.google.com */
@import url('https://fonts.googleapis.com/css?family=Open+Sans:400,700');
@import url('https://fonts.googleapis.com/css?family=Merriweather:400,700');

body {
  background-color: white;
  font-family: "Open Sans", sans-serif;
  padding: 5px 25px;
  font-size: 18px;
  margin: 0;
  color: #444;
}

h1 {
  font-family: "Merriweather", serif;
  font-size: 32px;
}

.test {
  color: red
}

I added a test.css to the test/one/ folder
.test {
   color: blue
}

And added a line to the index.html in the sub folder to read a one.css from the current directory (test/one/):
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Test relative addressing</title>
    <link rel="stylesheet" href="one.css">
  </head>
  <body>
    <p class="test">This should be blue</p>
  </body>
</html>

However in the Thimble Live Preview the text is not red.
I can see a 404 in the Network panel in Chrome for {url of root of project}/one.css
And then a fetch from the service worker for {url of root of project}/one.css which again returns 404.

This appears to be because the Live preview seems to create an iframe with a href="{url of root of project}/test/one/index.html" but in the <head> of that page it adds a <base href="{url of root of project)"> so that all links in this page in a sub folder becomes relative to the root of the project, not the current folder
<html data-brackets-id='11'>
  <head data-brackets-id='12' data-bramble-start-line='2' data-bramble-start-ch='2' data-bramble-end-line='6' data-bramble-end-ch='9'>
<base href="https://mozillathimblelivepreview.net/bramble/production/dist/vfs/.anonymous/projects/c87cdb1b-0b9a-4244-96f0-59565975a976/">
<script>
The link element has been written as 
<link data-brackets-id="24" data-bramble-start-line="5" data-bramble-start-ch="4" data-bramble-end-line="5" data-bramble-end-ch="42" rel="stylesheet" href="one.css">

  So it would be expected that it would try to resolve one.css from the project root.

  When the link elements are added the href is relative to this base href, not the current window.location

  If I live edit the href attribute in the link element to test/one/one.css the stylesheet is loaded and the elements render as expected.

  If I wanted to use relative addressing in test/one/index.html to reference the style.css at the root I'd add a link element such as 
<link rel="stylesheet" href="style.css">

  If I edit the html in thimble to use 
<link rel="stylesheet" href="test/one/one.css">

  Then it previews, finds the stylesheet and renders the elements with the CSS applied.

  If I add the following link element to the page in the sub folder test/one:
    <link rel="stylesheet" href="../../style.css">
  Then it tries to fetch the stylesheet from https://mozillathimblelivepreview.net/bramble/production/dist/vfs/.anonymous as again the href ../../style.css is relative to the base href of https://mozillathimblelivepreview.net/bramble/production/dist/vfs/.anonymous/projects/c87cdb1b-0b9a-4244-96f0-59565975a976/, ie https://mozillathimblelivepreview.net/bramble/production/dist/vfs/.anonymous/style.css

  However when i hover over the href in the link elements above in Chrome Dev. Tools. the URL it resolves to appears to be correct, but that's not where the css files get requested from

**Comment by gideonthomas** [comment]
Yeah I think the `base` tag should reference the current folder instead of the root folder. @humphd is that correct.

**Comment by ppatel221** [comment]
Hey guys, Id like to work on this. Can i get started on it?

**Comment by gideonthomas** [comment]
Hi @ppatel221, I might have prematurely marked this as a Good First Bug. Let's wait for a little guidance from @humphd first and if he agrees, we can help you figure out where to make the change.

**Comment by humphd** [comment]
We set the `<base>` element's URL here: https://github.com/mozilla/brackets/blob/master/src/extensions/default/bramble/lib/PostMessageTransport.js#L210.  If you want to have it match the current page's depth, we'd have to do some work in that function to figure out the relative path, and adjust the absolute URL accordingly.

We need to have a `<base>` for the case of Blob URLs needing to resolve relative links, so we can't dump this.  Whatever we do here in the preview, it needs to match what S3 will do when it's eventually published--will the case described above work properly on S3?

**Comment by gideonthomas** [comment]
Yeah S3 will handle this.
Would we set the base tag to the dirname of the current path?

**Comment by humphd** [comment]
We can try that approach, and see how it works.  Testing this in various browsers will be important, to make sure things work the way we expect.

**Comment by gideonthomas** [comment]
@ppatel221 would you like to / are you working on this?

**Comment by ArchitaDesai** [comment]
It's been over a month since @ppatel221 has commented and there's been no update afterward. So I would wanna get started on this as my first issue :grinning:

**Comment by gideonthomas** [comment]
@ArchitaDesai, sorry for the delays here, if you're still interested, feel free to work on this

**Comment by jinxac** [comment]
I would like to do this. Can you please tell how can i contribute?

**Comment by jwusteman** [comment]
Hi there - I am really anxious that this issue is sorted before my students start working on Thimble in September. It completely confuses them! How is it going? Thank you

**Comment by Niki24** [comment]
I have a group of students using Thimble as well. I was wondering if there was a solution other than moving the file outside of the folder. With my students being beginners, this was a frustrating find. I have my students download their projects so the base url wouldn't work in our case. This is an awesome web app. Let me know how I can help.

**Comment by twisted-sres** [comment]
Can I work on this as my first issue?

**Comment by jwusteman** [comment]
Hi there - Many thanks for taking this on! I am now even more anxious that this issue is sorted before my students start working on Thimble in September. It completely confuses them! How is it going?
